### PR TITLE
allow mce console to be disabled when ocp console nonexistent

### PR DESCRIPTION
**Pull request by eemurphy**
Signed-off-by: Erin Murphy <erinmurp@redhat.com>

Issue: https://github.com/stolostron/backlog/issues/25560

### CODE REVIEW

File: controllers/common.go [hunk 1]
```diff
@@ -20,11 +20,11 @@ import (
 
 	mcev1 "github.com/stolostron/backplane-operator/api/v1"
 
-	searchv2v1alpha1 "github.com/stolostron/search-v2-operator/api/v1alpha1"
-
 	configv1 "github.com/openshift/api/config/v1"
 	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
 	subv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	searchv2v1alpha1 "github.com/stolostron/search-v2-operator/api/v1alpha1"
+	slice "golang.org/x/exp/slices"
```
Comment (JakobGray) [inline controllers/common.go:27]: Golang describes the packages in `exp` as "experimental and unreliable" so I don't think we can rely on them.

Comment (JakobGray) [reply controllers/common.go:27]: https://github.com/stolostron/multiclusterhub-operator/blob/main/pkg/utils/utils.go#L444 might be able to replace the use of slice.Contains

**Review by JakobGray — COMMENTED** [review]
Will need to look at this more tomorrow, but one of the things we have specified as a criteria is:
> if OCP console is activated post-install, then MCE/ACM responds to the change and installs console (unless the component is disabled)

 Do we currently have a means of responding to a change in the OCP console activation?

**Comment by eemurphy** [comment]
@JakobGray If I'm correct, at L218 in mch-controller reconcile method, I have `result, err = r.setDefaults(multiClusterHub, ocpConsole)` . I assumed this gets called during reconcile, and will try to update components if any changes are made, such as ocpConsole being passed in as true after previously being false.

**Comment by JakobGray** [comment]
> @JakobGray If I'm correct, at L218 in mch-controller reconcile method, I have `result, err = r.setDefaults(multiClusterHub, ocpConsole)` . I assumed this gets called during reconcile, and will try to update components if any changes are made, such as ocpConsole being passed in as true after previously being false.

The changes will be applied when the reconcile is called, but the question is how long will that take? When talking about [watching a resource](https://github.com/stolostron/backlog/issues/25350#issuecomment-1238165396) we mean literally adding a watch on the resource so our reconciler responds to changes in it. Otherwise it could be hours until the next reconcile loop happens.

We define our watches when setting up the reconciler: https://github.com/stolostron/multiclusterhub-operator/blob/main/controllers/multiclusterhub_controller.go#L638. The goal is to only trigger a reconcile when it is actually relevant to us so we aren't wasting cycles. If it would consume too many resources or trigger too many reconciles it sometimes might not be worth it to add the watch.

**Comment by eemurphy** [comment]
> > @JakobGray If I'm correct, at L218 in mch-controller reconcile method, I have `result, err = r.setDefaults(multiClusterHub, ocpConsole)` . I assumed this gets called during reconcile, and will try to update components if any changes are made, such as ocpConsole being passed in as true after previously being false.
> 
> The changes will be applied when the reconcile is called, but the question is how long will that take? When talking about [watching a resource](https://github.com/stolostron/backlog/issues/25350#issuecomment-1238165396) we mean literally adding a watch on the resource so our reconciler responds to changes in it. Otherwise it could be hours until the next reconcile loop happens.
> 
> We define our watches when setting up the reconciler: https://github.com/stolostron/multiclusterhub-operator/blob/main/controllers/multiclusterhub_controller.go#L638. The goal is to only trigger a reconcile when it is actually relevant to us so we aren't wasting cycles. If it would consume too many resources or trigger too many reconciles it sometimes might not be worth it to add the watch.

I was working on a new watch but just realized that this watch on the ClusterVersion may work. Just to make sure I'm correct, will this watch here https://github.com/stolostron/multiclusterhub-operator/blob/main/controllers/multiclusterhub_controller.go#L671 trigger a reconcile any time a field in ClusterVersion struct changes? If so then we are ok, since now we need to watch `ClusterVersion.Status.Capabilities.EnabledCapabilities`

**Pull request closed without merging**